http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -124,10 +136,11 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
     for {
       ctx <- ev.ask
 
-      resourcePolicies <- samDao
-        .getResourcePolicies[R](authHeader)
-      res = resourcePolicies.filter { case (_, pn) =>
-        sr.policyNames.contains(pn)
+      resourcePolicies <- resources.toList.flatTraverse(resource =>
```

Comment:
It looks like this is now calling Sam `getResourcePolicies` once per resource which doesn't seem correct -- seems we want to call it once per resource _type_ ?

Reply:
Oh shoot, you are completely right! I was so focused on getting the compiler to work that I missed this, sorry!